File: solutions/javascript/secret-handshake/1/secret-handshake.js
//
// This is only a SKELETON file for the 'Secret Handshake' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

const actions = {
  4: 'wink',
  3: 'double blink',
  2: 'close your eyes',
  1: 'jump',
  0: 'reverse',
}

export const commands = (number) => {
  let res = []
  let binary = number.toString(2).padStart(5, 0);
  [...binary].forEach((item, index) => {
    if (item === '1') {
      res.unshift(actions[index])
    }
  })
  if (res.includes('reverse')) {
    res.splice(res.indexOf('reverse'), 1)
    res.reverse()
  }
  
  return res
};

File: solutions/haskell/secret-handshake/1/src/SecretHandshake.hs
module SecretHandshake (handshake) where

import Data.Bits ( Bits(testBit) )

handshake :: Int -> [String]
handshake n = if testBit n 4 then reverse h else h
    where
        h = [a | (a, b) <- actions, testBit n b]
        actions = zip ["wink", "double blink", "close your eyes", "jump"] [0..]